cmd/minikube/cmd/delete.go

```diff
@@ -601,49 +602,120 @@ func killMountProcess() error {
 	return nil
 }
 
+// killProcess takes a path to look for a pidfile (space-separated),
+// it reads the file and converts it to a bunch of pid ints,
+// then it tries to kill each one of them.
+// If no errors were ecnountered, it cleans the pidfile
 func killProcess(path string) error {
 	pidPath := filepath.Join(path, constants.MountProcessFileName)
 	if _, err := os.Stat(pidPath); os.IsNotExist(err) {
 		return nil
 	}
-
 	klog.Infof("Found %s ...", pidPath)
-	out, err := os.ReadFile(pidPath)
+
+	ppp, err := GetPids(pidPath)
 	if err != nil {
-		return errors.Wrap(err, "ReadFile")
+		return err
 	}
-	klog.Infof("pidfile contents: %s", out)
-	pid, err := strconv.Atoi(string(out))
-	if err != nil {
-		return errors.Wrap(err, "error parsing pid")
+
+	// we're trying to kill each process, without stopping at first error encountered
+	// error handling is done below
+	var errs []error
+	for _, pp := range ppp {
+		err := trySigKillProcess(pp, doesPIDBelongToMinikube)
+		if err != nil {
+			errs = append(errs, err)
+		}
+
 	}
-	// os.FindProcess does not check if pid is running :(
-	entry, err := ps.FindProcess(pid)
+
+	if len(errs) == 1 {
+		// if we've encountered only one error, we're returning it:
+		return errs[0]
+	} else if len(errs) != 0 {
+		// if multiple errors were encountered, combine them into a single error
+		out.Styled(style.Failure, "Multiple errors encountered:")
+		for _, e := range errs {
+			out.Err("%v\n", e)
+		}
+		return errors.New("multiple errors encountered while closing mount processes")
+	}
+
+	// if no errors were encoutered, it's safe to delete pidFile
+	if err := os.Remove(pidPath); err != nil {
+		return errors.Wrap(err, "While closing mount-pids file")
+	}
+
+	return err
+}
+
+// trySigKillProcess takes a PID as argument and tries to SIGKILL it.
+// It performs the OCHECK ownership check of the pid,
+// before trying to send a sigkill signal to it
+func trySigKillProcess(pid int, ocheck func(int) (bool, error)) error {
```

Comment:
I see you're trying to mock `doesPIDBelongToMinikube`.

What you can do instead is:
```
var doesPIDBelongToMinikube = func(pid int) (bool, error) {
```

Then during the unit test you can do
```
doesPIDBelongToMinikube = func(pid int) (bool, error) { return true, nil }
```

Reply:
got it, thanks :)